broadcast/src/buffered/engine.rs

```diff
@@ -97,9 +97,9 @@ impl<
 {
     /// Creates a new engine with the given context and configuration.
     /// Returns the engine and a mailbox for sending messages to the engine.
-    pub fn new(context: E, cfg: Config<Cfg, P>) -> (Self, Mailbox<P, D, M>) {
+    pub fn new(context: E, cfg: Config<Cfg, P>) -> (Self, Mailbox<Cfg, P, D, M>) {
```

Comment:
I think we should start calling `CodecConfig`/`Cfg` something more descriptive/type related.

I made this change when refactoring `consensus` and found it to be much clearer (otherwise it looks like the config to some global config).